### PR TITLE
Refactored KuraException and KuraRuntimeException

### LINKED ISSUE #817: Fix Exception Javadoc
In `KuraException` and `KuraRuntimeException`, `KuraErrorCode` has not been updated and it is mentioned as `EdcErrorCode`. Furthermore, `KuraException` is mentioned as `EdcException`.

### PR DESCRIPTION
**Pull request by amitjoy**
1. Modified Copyright Headers
2. Corrected, modified and added Javadoc comments
3. Removed dead codes
4. Minor changes in code styles
5. Added proper ordering of methods (Hence it shows complete git diff)
6. No method behaviour changes

Fixes #817

### CODE REVIEW

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java [hunk 1]
```diff
@@ -20,167 +18,191 @@
...
+public class KuraException extends Exception {
+
+    /** The Constant denoting resource bundle. */
+    private static final String KURA_EXCEPTION_MESSAGES_BUNDLE = "org.eclipse.kura.core.messages.KuraExceptionMessagesBundle";
+
+    /** The Constant denoting message pattern. */
+    private static final String KURA_GENERIC_MESSAGES_PATTERN = "Generic Error - {0}: {1}";
+
+    /** The Logger. */
+    private static final Logger s_logger = LoggerFactory.getLogger(KuraException.class);
```
Comment (ctron) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java]: As we are using the new code style, please use `logger`.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java [hunk 2]
```diff
@@ -20,167 +18,191 @@
...
+
+    /** The Constant denoting serial version identifier. */
+    private static final long serialVersionUID = 7468633737373095296L;
+
+    /** The associated arguments. */
+    private transient Object[] arguments;
```
Comment (ctron) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java]: Why are the arguments `transient`? I think we only should allow `Serializable` in general.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java [hunk 3]
```diff
@@ -20,167 +18,191 @@
...
+     *            the arguments
+     */
+    public KuraException(final KuraErrorCode code, final Throwable cause, final Object... arguments) {
+        super(cause);
+        this.code = code;
+        this.arguments = arguments;
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing
```
Comment (ctron) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java:91]: There should be at least a comment to not use this one. As we agreed on not using "INTERNAL_ERROR" anymore.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java [hunk 4]
```diff
@@ -20,167 +18,191 @@
...
+     * @param message
+     *            the message
+     * @return the kura exception
+     */
+    public static KuraException internalError(final String message) {
+        return new KuraException(KuraErrorCode.INTERNAL_ERROR, null, message);
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing a
```
Comment (ctron) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java:106]: There should be at least a comment to not use this one. As we agreed on not using "INTERNAL_ERROR" anymore.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java [hunk 5]
```diff
@@ -20,167 +18,191 @@
...
+     * @param cause
+     *            the cause
+     * @return the kura exception
+     */
+    public static KuraException internalError(final Throwable cause) {
+        return new KuraException(KuraErrorCode.INTERNAL_ERROR, cause, "");
+    }
+
+    /**
+     * Factory method to build an {@link KuraException} with the {@link KuraErrorCode#INTERNAL_ERROR} code providing a
```
Comment (ctron) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java:121]: There should be at least a comment to not use this one. As we agreed on not using "INTERNAL_ERROR" anymore.

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java [hunk 6]
```diff
@@ -20,167 +18,191 @@
...
+     * @return the error code
+     */
+    public KuraErrorCode getCode() {
+        return this.code;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getLocalizedMessage() {
+        return this.getLocalizedMessage(Locale.getDefault());
```
Comment (ctron) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java]: I don't think we require a `this.` here for method calling. Also not from our code style?!

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java [hunk 7]
```diff
@@ -20,167 +18,191 @@
...
+
+    /**
+     * Gets the localized message.
+     *
+     * @param locale
+     *            the locale
+     * @return the localized message
+     */
+    private String getLocalizedMessage(final Locale locale) {
+        final String pattern = this.getMessagePattern(locale, this.code);
```
Comment (ctron) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java]: I don't think we require a `this.` here for method calling. Also not from our code style?!

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java (hunk 3 above)
Comment (MMaiero) [reply kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java:91]: As agreed in https://github.com/eclipse/kura/issues/714#issuecomment-259356409 I would write that the usage of "INTERNAL_ERROR" is **highly discouraged**.

**Comment by amitjoy** [comment]
@ctron @MMaiero Requested changes are amended.

**Comment by ctron** [comment]
Could you please rebase so that the conflicts are resolved. :+1:  for merging the changes then.

**Comment by amitjoy** [comment]
@ctron Done 👍.

**Pull request merged**